### PR TITLE
Make queryset scoping customizable

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #2114
Required PR: https://github.com/pulp/pulp_file/pull/717

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/access_policy.py [hunk 1]
```diff
@@ -61,12 +60,17 @@ def handle_creation_hooks(cls, obj):
...
-        permission_name = getattr(view, "queryset_filtering_required_permission", None)
-        if permission_name:
-            user = view.request.user
-            qs = get_objects_for_user(user, permission_name, qs)
+        if access_policy := self.get_access_policy(view):
+            if access_policy.queryset_scoping:
+                scope = access_policy.queryset_scoping["function"]
+                if not (function := getattr(view, scope, None)):
+                    raise APIException(
+                        f"Queryset scoping method {scope} is not present on this view set."
```
[the remark this answers is not in the document]
Comment (gerrod3) [reply pulpcore/app/access_policy.py:70]: There's another exception in a method above that isn't using gettext. I'm still confused what strings should be wrapped.

File: pulpcore/app/viewsets/base.py [hunk 2]
```diff
@@ -258,6 +260,11 @@ def is_master_viewset(cls):
 
         return False
 
+    @property
+    def routable(self) -> bool:
```
Comment (ipanova) [inline pulpcore/app/viewsets/base.py:264]: should not you as part of this pr also override this property for list repos/distribution/publication viewsets?

**Comment by ipanova** [comment]
@gerrod3 looks good! :ok_hand:

**Review by bmbouter — APPROVED** [review]
Thanks! This looks great.

**Review by ipanova — APPROVED** [review]
Thank you!

**Pull request merged**